docs/sphinx/tutorial.rst

```diff
@@ -0,0 +1,68 @@
+.. ######################################################################################
+   # Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+   # See the top-level LICENSE file for details.
+   #
+   # SPDX-License-Identifier: BSD-3-Clause
+   ######################################################################################
+
+==============================
+CARE: CHAI and RAJA Extensions
+==============================
+
+Tutorial
+========
+
+CARE provides an abstraction layer built on top of CHAI and RAJA. Like RAJA, CARE has a set of execution policies, but these are simplified. These policies are the following: sequential, openmp, gpu, parallel, gpu_simulation, raja_chai_everywhere, raja_fusible, and raja_fusible_seq. One of the main difference between these policies and RAJA's policies is that the openmp and gpu policies fallback to the sequential policy if OpenMP or CUDA/HIP is not available. The parallel policy first attempts to use the GPU if available, then falls back to OpenMP, then falls back to the sequential policy. The main benefit of this approach is that clang-query can be used to detect what would be illegal memory accesses on the device using a regular CPU build.
+
+It should also be noted that there is a gpu_simulation mode. It is generally much harder to debug issues on the device, so we have a mode that runs the code on the host, and if CHAI is configured with -DCHAI_ENABLE_GPU_SIMULATION_MODE=ON, the "GPU" memory space is actually on the host as well. This enables some stale data to be caught more easily when CHAI is used as a memory manager, and also enables address sanitizer or valgrind to be used on the simulated GPU memory space. CARE must also be configured with -DCARE_ENABLE_GPU_SIMULATION_MODE=ON.
+
+Another policy of note is raja_chai_everywhere. This runs the code on both the host and the device and is intended to be used with managed_ptr to keep both the host and device object instance in sync.
+
+Finally, there are fusible policies. Loops can be "fused" together into a single kernel to reduce kernel launch overhead. More details will be forthcoming on this.
+
+These policies can be used with care::forall, which then forwards the arguments to RAJA::forall. CARE also provides convenient macros that can fall back to vanilla for loops if CARE is configured with -DENABLE_LEGACY_CARE=ON. These macros generally start with CARE and end with LOOP. For example, CARE_SEQUENTIAL_LOOP always executes sequentially. CARE_STREAM_LOOP executes on the GPU if GPU_ACTIVE is defined in the compilation unit and nvcc is used to compile the code, otherwise it falls back to executing using OpenMP on the host if OPENMP_ACTIVE is defined in the compilation unit and the correct OpenMP flags are passed to the compiler, otherwise it falls back to running sequentially on the host.
```

Comment:
Just change "GPU_ACTIVE is defined in the compilation unit and nvcc is used" to "nvcc and hipcc", other than that looks fine.

Reply:
Fixed.